server/lib/email.js

```diff
@@ -320,11 +320,17 @@ const generateEmailFromTemplateAndSend = async (template, recipient, data, optio
     return;
   }
 
-  const notificationIsActive = await models.Notification.isActive(template, data.collective.id, data.user.id);
-
-  if (!notificationIsActive) {
-    logger.info(`Email with template '${template}' not sent. Recipient email notification not active.`);
-    return;
+  // Unit test like collective.model.test.js don't care about sending
+  // the email only making sure the function was called
+  // hence data.collective or data.user are usually an empty object
+  // we skip the notification active check in this case.
+  if (process.env.NODE_ENV === 'production' && data.collective && data.user) {
```

Comment:
I would not rely on `data.collective`  and `data.user`.

I would just check `config.env === 'circleci` || config.env === 'test'. If it's not precise enough, I would make sure unit tests can be properly detected.

`config.env` is preferred to `process.env.NODE_ENV` but is usually the same value.

Reply:
Thinking of using this condition, only a single test is failing the notification active check because of the incomplete data structure.
```js
if (config.env === 'test' || config.env === 'circleci') {
    const user = { id: 1 };
    data.user = user;
    data.collective.id = 1;
  }
```
But I'm not completely comfortable with it.

Reply:
@flickz please commit what you think is best now, and we'll discuss it from there.

Reply:
@znarf The best possible way I think right now is to skip the unit test for active notification check.